velox/functions/prestosql/aggregates/tests/SumTest.cpp

```diff
@@ -13,17 +13,59 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+#include "velox/common/base/tests/GTestUtils.h"
 #include "velox/exec/AggregationHook.h"
+#include "velox/exec/tests/utils/Cursor.h"
 #include "velox/exec/tests/utils/PlanBuilder.h"
 #include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
 
 using facebook::velox::exec::test::PlanBuilder;
+using namespace facebook::velox::exec::test;
 
 namespace facebook::velox::aggregate::test {
 
 namespace {
 
-class SumTest : public AggregationTestBase {};
+class SumTest : public AggregationTestBase {
+ protected:
+  template <typename InputType, typename ResultType>
+  void testInputTypeLimits(bool expectOverflow = false) {
+    std::vector<std::vector<InputType>> inputs = {
+        {std::numeric_limits<InputType>::min(),
+         std::numeric_limits<InputType>::min() + 1},
+        {std::numeric_limits<InputType>::max(),
+         std::numeric_limits<InputType>::max() - 1}};
+    for (auto& input : inputs) {
+      std::vector<RowVectorPtr> data;
+      data.push_back(makeRowVector({makeFlatVector<InputType>(input)}));
+
+      // Testing these two steps provides enough coverage. Adding kfinal
+      // involves more elaborate multi-fragment setup which would be an
```

Comment:
It is not necessary to setup multi-fragment query to test final aggregation. Would you consider using AggregationTestBase::testAggregations instead?

Reply:
Thank you for pointing me to AggregationTestBase::testAggregations() this looks super useful to exhaustively test all aggregation node combinations.
For this test however it might be an overkill since we would only want to make sure that overflow gets caught for each step individually and chaining steps wont provide additional coverage.

To provide more context for my code comment above: I wanted to test the kfinal step with an input of 2 values that when added cause an overflow. (Please correct me if I am wrong) Since this is a global aggregation any previous aggregation (partial) step that I add to the same query fragment would result in hitting the overflow in that step itself. Therefore to make sure that i hit the overflow in the kFinal step the only way I could think of was to create 2 separate input fragments that both output a single value, which when aggregates at the kFinal step and hits an overflow.

Something akin to this looks promising but seems like its only run when grouping keys exist: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/aggregates/tests/AggregationTestBase.cpp#L176-L196
For my test case specifically I can mimic the same steps above and try putting the 2 input values in 2 separate rowVectors and test whether they get plugged in through separate drivers feeding into a final agg. I'll report back if that works.